flang/lib/Optimizer/Transforms/AffinePromotion.cpp

```diff
@@ -312,18 +312,29 @@ mlir::Type coordinateArrayElement(fir::ArrayCoorOp op) {
 std::pair<mlir::AffineApplyOp, fir::ConvertOp>
 createAffineOps(mlir::Value arrayRef, mlir::PatternRewriter &rewriter) {
   auto acoOp = arrayRef.getDefiningOp<ArrayCoorOp>();
-  auto genDim = acoOp.dims().getDefiningOp<GenDimsOp>();
+  auto genDim = acoOp.getShape().getDefiningOp<ShapeOp>();
   auto affineMap =
-      createArrayIndexAffineMap(acoOp.coor().size(), acoOp.getContext());
+      createArrayIndexAffineMap(acoOp.getIndices().size(), acoOp.getContext());
   SmallVector<mlir::Value, 4> indexArgs;
-  indexArgs.append(acoOp.coor().begin(), acoOp.coor().end());
-  indexArgs.append(genDim.triples().begin(), genDim.triples().end());
+  indexArgs.append(acoOp.getIndices().begin(), acoOp.getIndices().end());
+
+  // FIXME: quick hack for now (assumes 1 for the shift and stride)
```

Comment:
Another possible way is to have separate implementations for `createArrayIndexAffineMap` in case of shape and slice. It would spread the complexity appropriately. I can look at it later after this is checked in.

Reply:
Yes, I admit this is a really bad Scotch tape job. It gets the tests to pass, but it's not what we want here.